core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -2749,13 +2749,12 @@ class KafkaApis(val requestChannel: RequestChannel,
       )
     }
 
-    val partitionsOpt = listPartitionReassignmentsRequest.data.topics match {
-      case topics: Any =>
-        Some(topics.iterator().asScala.flatMap { topic =>
-          topic.partitionIndexes.iterator().asScala
-            .map { tp => new TopicPartition(topic.name(), tp) }
-        }.toSet)
-      case _ => None
+    val partitionsOpt = Option(listPartitionReassignmentsRequest.data.topics).map { topics =>
+      topics.iterator().asScala.flatMap { topic =>
+        topic.partitionIndexes.iterator().asScala.map { tp =>
```

Comment:
nit: usually `tp` is reserved for `TopicPartition` instances, which makes this code look a bit funny. Perhaps we could rename the variable to `partitionIndex`?

Reply:
I agree. Fixed.